src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -1148,10 +1148,72 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
             case 2:
                 op2 = getArgForHWIntrinsic(sigReader.GetOp2Type(), sigReader.op2ClsHnd);
                 op2 = addRangeCheckIfNeeded(intrinsic, op2, mustExpand, immLowerBound, immUpperBound);
-                op1 = getArgForHWIntrinsic(sigReader.GetOp1Type(), sigReader.op1ClsHnd);
 
-                retNode = isScalar ? gtNewScalarHWIntrinsicNode(retType, op1, op2, intrinsic)
-                                   : gtNewSimdHWIntrinsicNode(retType, op1, op2, intrinsic, simdBaseJitType, simdSize);
+#ifdef TARGET_ARM64
+                if ((intrinsic == NI_AdvSimd_VectorTableLookup) || (intrinsic == NI_AdvSimd_Arm64_VectorTableLookup))
+                {
+                    op1 = impPopStack().val;
+
+                    if (op1->TypeGet() == TYP_STRUCT)
+                    {
+                        assert(op1->OperIsLocal());
+
```

Comment:
Can we actually assume `op1` is a local here? I would expect not, e. g. it could be an indir or a call. These cases would need to be handled by spilling a local (via `impAssignTempGen`).

Also, it would seem cleaner to me to put this intrinsic in the "requires special import" category and handle it in `hwintrinsicarm64.cpp`/`impSpecialIntrinsic` instead of here.

Reply:
Done.